src/config/cors_params.rs

```diff
@@ -0,0 +1,22 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq, Eq, schemars::JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct CorsParams {
+    pub allow_credentials: Option<bool>,
+    pub allow_headers: ConstOrMirror,
+    pub allow_methods: ConstOrMirror,
+    pub allow_origin: Vec<String>,
+    pub allow_private_network: bool,
+    pub expose_headers: Option<String>,
+    pub max_age: Option<usize>,
+    pub vary: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq, Eq, schemars::JsonSchema)]
+#[serde(untagged)]
```

Comment:
is there specific purpose for untagged here? looking into json schema it looks strange actually

Reply:
I think this will allow developers to write the config as 

```json
{
  allowHeaders: "*"
}
```
and also as 

```json
{
  allowHeaders: ["x-foo", "x-bar"]
}
`